templates/rag-conversation-zep/rag_conversation_zep/chain.py

```diff
@@ -0,0 +1,149 @@
+import os
+from operator import itemgetter
+from typing import List, Tuple
+
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain.prompts.prompt import PromptTemplate
+from langchain.schema import AIMessage, HumanMessage, format_document
+from langchain.schema.output_parser import StrOutputParser
+from langchain.schema.runnable import (
+    RunnableBranch,
+    RunnableLambda,
+    RunnableMap,
+    RunnablePassthrough,
+)
+from langchain.vectorstores.zep import CollectionConfig, ZepVectorStore
+from pydantic import BaseModel, Field
+
+ZEP_API_URL = os.environ.get("ZEP_API_URL", "http://localhost:8000")
+ZEP_API_KEY = os.environ.get("ZEP_API_KEY", None)
+ZEP_COLLECTION_NAME = os.environ.get("ZEP_COLLECTION", "langchaintest")
+
+# Zep offers native, hardware-accelerated MMR. Enabling this will improve
+# the diversity of results, but may also reduce relevance. You can tune
+# the lambda parameter to control the tradeoff between relevance and diversity.
+# Enabling is a good default.
+ZEP_MMR = True
+ZEP_MMR_LAMBDA = 0.5
+
+collection_config = CollectionConfig(
+    name=ZEP_COLLECTION_NAME,
+    description="Zep collection for LangChain",
+    metadata={},
+    embedding_dimensions=1536,
+    is_auto_embedded=True,
+)
+
+
+# ## Ingest code - you may need to run this the first time
```

Comment:
maybe we put this in a separate ingest.py file outside the package, since it's not really part of the application? similar to rag-elastic search